research/picai/picai_server.py

```diff
@@ -0,0 +1,164 @@
+import timeit
+from logging import INFO
+from pathlib import Path
+from typing import Optional
+
+import torch.nn as nn
+from flwr.common.logger import log
+from flwr.server.client_manager import ClientManager
+from flwr.server.history import History
+from flwr.server.strategy import Strategy
+
+from fl4health.checkpointing.checkpointer import ServerPerRoundCheckpointer, TorchCheckpointer
+from fl4health.parameter_exchange.parameter_exchanger_base import ParameterExchanger
+from fl4health.reporting.fl_wanb import ServerWandBReporter
+from fl4health.server.base_server import FlServerWithCheckpointing
+from research.picai.fl_utils import get_initial_model_parameters
+
+
+class PicaiServer(FlServerWithCheckpointing):
```

Comment:
I think this is a very nice extension of `FlServerWithCheckpointing`!  I was wondering if there is a way to have this included in the core library as a pre-emption resistant server (either in this PR or in a follow-up ticket).

Reply:
Thanks for the great feedback! I will definitely have a follow up PR to incorporate this into the main. I decided not to do that for this PR, just to not tackle too much at once